src/ocm/components/hosts/HostsDiscoveryTable.tsx

```diff
@@ -51,7 +51,7 @@ const HostsDiscoveryTable: React.FC<HostsDiscoveryTableProps> = ({
 
   const content = React.useMemo(
     () => [
-      hostnameColumn(onEditHost),
+      hostnameColumn(onEditHost, undefined, !actionChecks.canEditHostname()),
```

Comment:
same here, just pass the callback
```suggestion
      hostnameColumn(onEditHost, undefined, actionChecks.canEditHostname),
```